File: backend/flight_booking_project/src/main/java/com/javaweb/flight_booking_project/dto/response/FlightResponse.java
package com.javaweb.flight_booking_project.dto.response;

import lombok.Data;

import java.util.Date;
import java.util.List;

@Data
public class FlightResponse {
    private Long id;
    private String codeFlight;
    private Date departureTime;
    private Date arrivalTime;
    private String departureAirport;
    private String departureLocation;
    private String arrivalLocation;
    private String arrivalAirport;
    private String codeDepartAirport;
    private String codeArriAirport;
    private String airline;
    private String logoAirline;
    private PlaneResponse plane;
    private int availableEco;
    private int availableBus;
    private double ecoPrice;
    private double busPrice;
    private Boolean status;
    private Long ticketId;
    List<SeatResponse> seats;
    List<LuggageResponse> luggages;

}

File: mobile/lib/data/model/flight_response.dart
import 'package:mobile/data/model/plane_response.dart';

class FlightResponse {
  int id;
  String codeFlight;
  DateTime? departureTime;
  DateTime? arrivalTime;
  String departureAirport;
  String departureLocation;
  String arrivalLocation;
  String arrivalAirport;
  String airline;
  String logoAirline;
  PlaneResponse plane;
  double ecoPrice;
  double busPrice;

  FlightResponse({
    required this.id,
    required this.codeFlight,
    required this.departureTime,
    required this.arrivalTime,
    required this.departureAirport,
    required this.departureLocation,
    required this.arrivalLocation,
    required this.arrivalAirport,
    required this.airline,
    required this.logoAirline,
    required this.plane,
    required this.ecoPrice,
    required this.busPrice,
  });

  factory FlightResponse.fromJson(Map<String, dynamic> json) {
    return FlightResponse(
      id: json['id'],
      codeFlight: json['codeFlight'],
      departureTime: json['departureTime'] != null
          ? DateTime.parse(json['departureTime'])
          : null,
      arrivalTime: json['arrivalTime'] != null
          ? DateTime.parse(json['arrivalTime'])
          : null,
      departureAirport: json['departureAirport'],
      departureLocation: json['departureLocation'],
      arrivalLocation: json['arrivalLocation'],
      arrivalAirport: json['arrivalAirport'],
      airline: json['airline'],
      logoAirline: json['logoAirline'],
      plane: PlaneResponse.fromJson(json['plane']),
      ecoPrice: json['ecoPrice'],
      busPrice: json['busPrice'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'codeFlight': codeFlight,
      'departureTime': departureTime,
      'arrivalTime': arrivalTime,
      'departureAirport': departureAirport,
      'departureLocation': departureLocation,
      'arrivalLocation': arrivalLocation,
      'arrivalAirport': arrivalAirport,
      'airline': airline,
      'logoAirline': logoAirline,
      'plane': plane,
      'ecoPrice': ecoPrice,
      'busPrice': busPrice,
    };
  }

  static List<FlightResponse> fromJsonList(List<dynamic> jsonList) {
    return jsonList.map((json) => FlightResponse.fromJson(json)).toList();
  }
}
